packages/fuels-core/src/types/bits_256.rs

```diff
@@ -50,19 +50,80 @@ impl Tokenizable for Bits256 {
     }
 }
 
+#[derive(Debug, PartialEq, Eq, Copy, Clone)]
+// ANCHOR: evm_address
+pub struct EvmAddress {
+    value: Bits256,
+}
+// ANCHOR_END: evm_address
+
+impl EvmAddress {
+    fn clear_12_bytes(bytes: [u8; 32]) -> [u8; 32] {
+        let mut bytes = bytes;
+        bytes[..12].copy_from_slice(&[0u8; 12]);
+
+        bytes
+    }
+}
+
+impl From<Bits256> for EvmAddress {
+    fn from(b256: Bits256) -> Self {
+        let value = Bits256(Self::clear_12_bytes(b256.0));
+
+        Self { value }
+    }
+}
+
+impl Parameterize for EvmAddress {
+    fn param_type() -> ParamType {
+        ParamType::Struct {
+            fields: vec![ParamType::B256],
+            generics: vec![],
+        }
+    }
+}
+
+impl Tokenizable for EvmAddress {
+    fn from_token(t: Token) -> Result<Self, Error>
+    where
+        Self: Sized,
+    {
+        if let Token::Struct(tokens) = t {
+            let first_token = tokens.into_iter().next();
+            if let Some(Token::B256(data)) = first_token {
+                Ok(EvmAddress::from(Bits256(data)))
+            } else {
+                Err(Error::InstantiationError(format!(
```

Comment:
What I've found is that these 'expected X got Y' are not that great to run into without a backtrace.

When I remember I try to say something like: "EvmAddress could not be constructed from the given token. Reason: Expected Struct(B256) got: {token:?}"